packages/near-membrane-base/src/red.ts

```diff
@@ -183,11 +178,16 @@ export const serializedRedEnvSourceText = (function redEnvFactory(blueEnv: Membr
     }
 
     function getDistortedValue(target: RedProxyTarget): RedProxyTarget {
-        if (!WeakMapHas(distortionMap, target)) {
-            return target;
+        let distortedTarget: RedProxyTarget = target;
+        try {
+            distortedTarget = distortionCallback(target);
+        } finally {
+            // if a distortion entry is found, it must be a valid proxy target
+            if (distortedTarget !== target && typeof distortedTarget !== typeof target) {
+                throw new ErrorCtor(`Invalid distortion ${target}.`);
+            }
```

Comment:
As a bonus later we can use the replacement plugin with something like `environment.DEV_MODE` which is replaced with `true` or `false` and have it remove the code chunk for prod (minified builds).